nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
I am not convinced as to why other caching systems are needed for what should be a light system - you've not really presented _why_ what we provide will not be sufficient - I see no use case. As a result, I really don't think this is necessary, **and we shouldn't support this.**

* Nucleus is an essentials plugin, and this is meant to be a relatively light system anyway that does the basics well. More sophisticated plugins should replace our system completely, rather than providing a different cache system.
* Many server owners will not understand what swapping out the caching system will mean, and will then come to Nucleus when tickets don't work because of a poorly implemented/setup caching system is in place. We cannot hope to debug that.

I do not think this should be here - heavier networks may be better off with deidicated ticketing systems (and I really do encourage alternative implementations). Indeed, I feel like this is the crux of the matter, you might want to use this as a (potentially) cross server system and you want the ability to invalidate caches when the ticket is written to from another location, possibly even a web browser. Whilst I agree we could provide the APIs to get/retrieve this data from the plugin (and _possibly_ the ability to invalidate the cache for a ticket from the ticket service), we should _not_ be providing a way to swap out a cache system because I think it's just going to cause more problems down the line.

Nucleus _is not_ a cross server/network plugin - and while I may take steps to share _some_ data, I do not think that it is within Nucleus' realm to provide cross server functionallity, that would be too much for this plugin and I think that we should be focussing on the per-server experience.